google/resource_compute_region_instance_group_manager.go

```diff
@@ -182,7 +190,38 @@ func resourceComputeRegionInstanceGroupManagerCreate(d *schema.ResourceData, met
 		return err
 	}
 
-	return resourceComputeRegionInstanceGroupManagerRead(d, meta)
+	c1 := make(chan error, 1)
+	c2 := make(chan bool, 1)
+	go func() {
+		for {
+			select {
+			case <-c2:
+				return
+			default:
+				err = resourceComputeRegionInstanceGroupManagerRead(d, meta)
+				if _, ok := err.(notFinishedCreatingError); ok {
+					time.Sleep(5 * time.Second)
+					continue
+				} else {
+					c1 <- err
+				}
+			}
+		}
+	}()
+	select {
```

Comment:
why using goroutine if you are blocking the main thread anyway here?

Reply:
In this case, because there are two goroutines - time.After() returns a channel that sends "true" after the duration specified.  This serves as an (apparently) idiomatic timeout mechanism.  I like it, but willing to reformulate it if you prefer.  Definitely need to comment it.

Reply:
You could probably rely on state.WaitForState() and wait until the creating + creatingwhitoutretries is equal to 0.

It would be more consistent with the rest of the waiting logic in Terraform

Reply:
Aha, yes, state.WaitForState is a much better version of what I've written here.  Will do!